gapic/src/main/com/google/gapid/perfetto/models/FrameEventsTrack.java

```diff
@@ -75,6 +77,19 @@ public static FrameEventsTrack forBuffer(QueryEngine qe, GpuInfo.Buffer buffer)
     return new FrameEventsTrack(qe, buffer.trackId);
   }
 
+  public static RGBA getColor(String title) {
+    return colorForSlice(title, 0);
+  }
+
+  public static RGBA getBorderColor(String title) {
+    return colorForSlice(title, StyleConstants.isLight() ? -5 : 5);
+  }
+
+  private static RGBA colorForSlice(String title, int shadeIdx) {
+    return StyleConstants.Palette.getColor(
+        (title.hashCode() ^ title.length()) & 0x7fffffff, shadeIdx);
+  }
+
```

Comment:
This should be in the view class, not the model.

Reply:
I took this mainly from the SliceTrack class which is inside models, which is why I put it in models too.